[CI] EsqlSpecIT test {mv_percentile.FromIndex SYNC} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3800 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/7r3d4bmhn4gck)
- [elasticsearch-intake #9022 / part3](https://gradle-enterprise.elastic.co/s/aovy3mnkir764)
- [elasticsearch-pull-request #28419 / part-3](https://gradle-enterprise.elastic.co/s/c3pxf4bixerrw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT" -Dtests.method="test {mv_percentile.FromIndex SYNC}" -Dtests.seed=2942AF53FC53FD80 -Dtests.locale=sk -Dtests.timezone=America/Fort_Wayne -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bmv_percentile.FromIndex%20SYNC%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Data mismatch:
row 1 column 2:a list containing
row 1 column 2:0: expected <-7.08> but was <-7.125>
row 2 column 0:a list containing
row 2 column 0:0: expected "-6" but was "-7"
row 2 column 1:a list containing
row 2 column 1:0: expected "-6" but was "-7"
row 2 column 2:a list containing
row 2 column 2:0: expected <-6.9> but was <-7.125>
```

**Issue Reasons:**
- [main] 3 failures in test test {mv_percentile.FromIndex SYNC} (8.1% fail rate in 37 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -176,12 +176,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT
   method: testScaledFloat
   issue: https://github.com/elastic/elasticsearch/issues/112003
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {mv_percentile.FromIndex SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/112036
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {mv_percentile.FromIndex ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/112037
 - class: org.elasticsearch.xpack.esql.qa.single_node.RestEsqlIT
   method: testForceSleepsProfile {SYNC}
   issue: https://github.com/elastic/elasticsearch/issues/112039

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/mv_percentile.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/mv_percentile.csv-spec
@@ -95,15 +95,15 @@ FROM employees
     integer = MV_PERCENTILE(salary_change.int, 75),
     long = MV_PERCENTILE(salary_change.long, 75),
     double = MV_PERCENTILE(salary_change, 75)
-| KEEP integer, long, double
+| KEEP emp_no, integer, long, double
 | SORT double
 | LIMIT 3
 ;
 
-integer:integer | long:long | double:double
--8              | -8        | -8.46
--7              | -7        | -7.08
--6              | -6        | -6.9
+emp_no:integer | integer:integer | long:long | double:double
+10034          | -8              | -8        | -8.46
+10037          | -7              | -7        | -7.08
+10039          | -6              | -6        | -6.9
 ;
 
 fromIndexPercentile

